pkg/operation/botanist/component/nginxingressshoot/nginxingress_test.go

```diff
@@ -0,0 +1,781 @@
+// Copyright (c) 2023 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package nginxingressshoot
+
+import (
+	"context"
+
+	"github.com/Masterminds/semver"
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	resourcesv1alpha1 "github.com/gardener/gardener/pkg/apis/resources/v1alpha1"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/pkg/operation/botanist/component"
+	"github.com/gardener/gardener/pkg/utils"
+	"github.com/gardener/gardener/pkg/utils/retry"
+	retryfake "github.com/gardener/gardener/pkg/utils/retry/fake"
+	"github.com/gardener/gardener/pkg/utils/test"
+	. "github.com/gardener/gardener/pkg/utils/test/matchers"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	fakeclient "sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+var _ = Describe("NginxIngress", func() {
+	var (
+		ctx = context.TODO()
+
+		managedResourceName  = "shoot-addon-nginx-ingress"
+		namespace            = "some-namespace"
+		nginxControllerImage = "some-image:some-tag"
+		defaultBackendImage  = "some-image2:some-tag2"
+		configMapData        = map[string]string{
+			"foo":  "bar",
+			"dot":  "3",
+			"dash": "false",
+		}
+
+		c         client.Client
+		component component.DeployWaiter
+
+		managedResource       *resourcesv1alpha1.ManagedResource
+		managedResourceSecret *corev1.Secret
+
+		values = Values{
+			NginxControllerImage: nginxControllerImage,
+			DefaultBackendImage:  defaultBackendImage,
+			ConfigData:           configMapData,
+			PSPDisabled:          true,
```

Comment:
It is better to have a default Kubernetes version here, so you don't need to set it for every test.

Reply:
No strong opinion. I have kept the Kubernetes version in separate context to better define the corresponding test cases and for better readability.